docs/JestObjectAPI.md

```diff
@@ -314,6 +314,46 @@ test('moduleName 2', () => {
 });
 ```
 
+Using `jest.doMock()` with ES6 modules requires additional steps:
+
+- We have to specify the `__esModule: true` property (see the [`jest.mock()`](#jestmockmodulename-factory-options) API for more information).
+- Static ES6 module imports are hoisted to the top of the file, so instead we have to import them dynamically using `import()`.
+- Finally, we need an environment which supports dynamic importing. Please see [Using Babel](GettingStarted.md#using-babel) for the initial setup. Then add the plugin [babel-plugin-dynamic-import-node](https://www.npmjs.com/package/babel-plugin-dynamic-import-node), or an equivalent, to your Babel config to enable dynamic importing in Node.
+
+```js
+beforeEach(() => {
+  jest.resetModules();
+});
+
+test('moduleName 1', () => {
+  jest.doMock('../moduleName', () => {
+    return {
+      __esModule: true,
+      default: 'default1',
+      foo: 'foo1',
+    };
+  });
+  import('../moduleName').then(moduleName => {
```

Comment:
Please either use async/await or return this promise from test – otherwise it's not gonna work properly. Same for the test below.

Reply:
Ah yes you're correct. Surprisingly these worked for me anyway (i.e. when changing the value in `toEqual` the test failed), possibly some implementation detail of mocking and Babel.

Changed.